src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,304 @@ void CmdSketcherViewSection::activated(int iMsg)
+    sWhatsThis = "Sketcher_GridToggle";
+    sStatusTip = sToolTipText;
+    eType = 0;
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    sPixmap = hGrp->GetBool("ShowGrid") ? "Sketcher_GridToggle" : "Sketcher_GridToggle_Deactivated";
+}
+
+void CmdSketcherGridToggle::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    bool newState = toggleGrid(getActiveGuiDocument());
+    getAction()->setIcon(Gui::BitmapFactory().iconFromTheme(newState ? "Sketcher_GridToggle" : "Sketcher_GridToggle_Deactivated"));
+}
+
+bool CmdSketcherGridToggle::isActive()
+{
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(getActiveGuiDocument()->getInEdit());
+    if (sketchView)
+        getAction()->setIcon(Gui::BitmapFactory().iconFromTheme(sketchView->ShowGrid.getValue() ? "Sketcher_GridToggle" : "Sketcher_GridToggle_Deactivated"));
+
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid Snap */
+DEF_STD_CMD_A(CmdSketcherGridSnap)
+
+bool toggleGridSnap(Gui::Document* doc) {
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    bool newState = !hGrp->GetBool("GridSnap"); //In case (!sketchView)
+
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    if (sketchView) {
+        newState = !sketchView->GridSnap.getValue();
+        sketchView->GridSnap.setValue(newState);
+    }
+    return newState;
+}
+
+CmdSketcherGridSnap::CmdSketcherGridSnap()
+    : Command("Sketcher_GridSnap")
+{
+    sAppModule = "Sketcher";
+    sGroup = "Sketcher";
+    sMenuText = QT_TR_NOOP("Grid Snap");
+    sToolTipText = QT_TR_NOOP("New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid size to a grid line to snap.");
+    sWhatsThis = "Sketcher_GridSnap";
+    sStatusTip = sToolTipText;
+    eType = 0;
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    sPixmap = hGrp->GetBool("GridSnap") ? "Sketcher_GridSnap" : "Sketcher_GridSnap_Deactivated";
+}
+
+void CmdSketcherGridSnap::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    bool newState = toggleGridSnap(getActiveGuiDocument());
+    getAction()->setIcon(Gui::BitmapFactory().iconFromTheme(newState ? "Sketcher_GridSnap" : "Sketcher_GridSnap_Deactivated"));
 }
 
+bool CmdSketcherGridSnap::isActive()
+{
+    auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(getActiveGuiDocument()->getInEdit());
+    if (sketchView)
+        getAction()->setIcon(Gui::BitmapFactory().iconFromTheme(sketchView->GridSnap.getValue() ? "Sketcher_GridSnap" : "Sketcher_GridSnap_Deactivated"));
+    
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid Auto */
+DEF_STD_CMD_A(CmdSketcherGridAuto)

```

Comment:
Same here. The command doesn't do anything meaningful and can be removed.

Reply:
Ok so I remove these 3 commands and just leave the comp.